### DISCUSSION TITLE
Orbit with Height specs

### DISCUSSION THREAD

**Discussion by gwbres** [nyx-space/anise#296] [Ideas]
Hello, I have a rare scenario that could be answered in the long term,
where I have an `Orbit` projected on solid body ground (CEF position) and I want to adjust its altitude (height in km) with my own specs.
Currently we have options to determine the height of a given orbit, but none to modify it. 
I have no idea how complex that can be.
It would be similar to the manual velocity specs that exist (`with_velocity`)

**Comment by ChristopherRabotin** [comment]
Hi Guillaume,

I don't think it would be too difficult to implement that. The simplest fashion would be to compute the lat/long using ANISE's built-in method, then have the `with_altitude_km` function require the angular velocity of the body, and re-initialize an orbit using the `try_latlongalt` function (code here: https://docs.rs/anise/latest/src/anise/astro/orbit_geodetic.rs.html#82-112).

Another option would be to recompute the angular velocity from the current velocity vector, but that may lead to rounding errors I think, I'm not sure.

Feel free to open a ticket on the repo as a new requested feature.
